internal/controller/drplacementcontrol_controller.go

```diff
@@ -2885,3 +2889,76 @@ func (r *DRPlacementControlReconciler) twoVMDRPCsConflict(drpc *rmn.DRPlacementC
 
 	return false
 }
+
+// ensureDoNotDeletePVCAnnotation ensures that the "do-not-delete-pvc" annotation is propagated from the DRPC
+// resource to the VRG resource on the specified cluster. If the annotation is set on the DRPC but not yet
+// present on the VRG, this function updates the VRG ManifestWork to include the annotation. This is used to
+// prevent deletion of app PVCs during DR disabling.
+func (r *DRPlacementControlReconciler) ensureDoNotDeletePVCAnnotation(
+	mwu rmnutil.MWUtil,
+	drpc *rmn.DRPlacementControl,
+	vrg *rmn.VolumeReplicationGroup,
+	cluster string,
+	log logr.Logger,
+) error {
+	if vrg.Spec.ReplicationState == rmn.Primary {
+		return nil // Only propagate the annotation if the VRG is in primary state
```

Comment:
Is it possible, that the comment and the code contradict each other? If state is primary nil is returned and no propagation happens.